### DISCUSSION TITLE
Custom Curve

### DISCUSSION THREAD

**Discussion by Gourishh** [community/community#131497] [Programming Help]
### Body

I am developing an App using flutter and came across a custom curve like structure and I have no Idea how to create one. I did some research and found that it might be possible with clip path widget but again I have no clue how will that work. I am sharing a screenshot for the same. 

![image](https://github.com/community/community/assets/115629874/9f950984-b2d6-47fd-84b8-34dc2c56508e)

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @Gourishh ,

I'm not a Dart/Flutter expert but take a look at this [thread](https://stackoverflow.com/questions/72693111/how-to-create-this-curves-in-flutter).

You can use the  [Stack](https://api.flutter.dev/flutter/widgets/Stack-class.html) widget for the background and the [CustomPaint](https://api.flutter.dev/flutter/widgets/CustomPaint-class.html) widget to paint custom shapes on the background. You can also use [FlutterShapeMaker](https://fluttershapemaker.com/#/) to shape your curve directly and export it to a [CustomPainter](https://api.flutter.dev/flutter/rendering/CustomPainter-class.html) class.

**Comment by steroSterka** [comment]
Hi @Gourishh, 
check this out: https://stackoverflow.com/questions/57123956/how-to-make-curved-app-bar-with-custom-clipper-in-flutter
or here for better step by step: https://medium.com/flutter-community/custom-shaped-appbar-as-seen-in-the-bunny-search-app-6312d067485c

**Comment by odelyo** [comment]
To create the custom curve-like structure you highlighted in your Flutter app, you can indeed use the ClipPath widget. The ClipPath widget allows you to clip widgets with custom shapes using a CustomClipper. Here's a simple guide to help you get started:

Step 1: Create a Custom Clipper
First, create a custom clipper class that defines the path for your curved shape. You can use the Path class to draw the custom shape.
```
import 'package:flutter/material.dart';

class CustomCurveClipper extends CustomClipper<Path> {
  @override
  Path getClip(Size size) {
    var path = Path();
    // Start at the top-left corner
    path.lineTo(0, 0);
    path.lineTo(0, size.height - 50);

    // Draw the curve
    var firstControlPoint = Offset(size.width / 4, size.height);
    var firstEndPoint = Offset(size.width / 2, size.height - 30);
    var secondControlPoint = Offset(3 * size.width / 4, size.height - 80);
    var secondEndPoint = Offset(size.width, size.height - 40);

    path.quadraticBezierTo(firstControlPoint.dx, firstControlPoint.dy,
        firstEndPoint.dx, firstEndPoint.dy);
    path.quadraticBezierTo(secondControlPoint.dx, secondControlPoint.dy,
        secondEndPoint.dx, secondEndPoint.dy);

    // Complete the path by closing it at the top-right corner
    path.lineTo(size.width, size.height - 40);
    path.lineTo(size.width, 0);

    path.close();
    return path;
  }

  @override
  bool shouldReclip(covariant CustomClipper<Path> oldClipper) {
    return false;
  }
}

```
Step 2: Apply the Custom Clipper with ClipPath
Now, you can use the ClipPath widget to apply your custom clipper to any widget, such as a Container with a background color or an image.
```
import 'package:flutter/material.dart';

class CustomCurveWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          ClipPath(
            clipper: CustomCurveClipper(),
            child: Container(
              height: 200,
              color: Colors.blue,
              child: Center(
                child: Text(
                  'Custom Curve',
                  style: TextStyle(
                    color: Colors.white,
                    fontSize: 24,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ),
            ),
          ),
          // Add your other widgets below the curved container
        ],
      ),
    );
  }
}

void main() {
  runApp(MaterialApp(
    home: CustomCurveWidget(),
  ));
}

```
